src/s3fs.cpp

```diff
@@ -3426,11 +3461,18 @@ static int s3fs_removexattr(const char* path, const char* name)
   meta["x-amz-copy-source"]        = urlEncode(service_path + bucket + get_realpath(strpath.c_str()));
   meta["x-amz-metadata-directive"] = "REPLACE";
 
-  if(0 != put_headers(strpath.c_str(), meta, true)){
-    free_xattrs(xattrs);
-    return -EIO;
+  // Since meta data cannot be updated during uploading, it puts with a delay.
```

Comment:
This change confuses me -- why doesn't locking prevent this situation?  During sync, upload should take the metadata lock, copy the headers it needs, take the data lock, release the metadata lock, then initiate upload.  Is my understanding incorrect?

Reply:
If user moves(mv) the file, Hooks of s3fs are called as follows.
1. s3fs_create
1. s3fs_write (called multiple times for each block size)
1. s3fs_utimene
1. s3fs_chown
1. s3fs_chmod
1. s3fs_flush
1. s3fs_release

If the disk space is safe, a call to s3fs_flush will start and end the multipart upload.
However, if you are running out of disk space, multipart upload is starting during s3fs_write is called multiple times.
And the multipart upload is completed at s3fs_flush.
Thus, during this multipart upload, mtime/mod/owner changes will be called.
If this is blocked(exclusive object: mutex etc), the process itself will stop and s3fs will deadlock.

Therefore, while executing(after starting) multipart upload, the updated contents of the meta data will be accumulated.
I think that the function that may call put_header during multipart upload needs this accumulation processing.